trade-finance-manager-ui/server/controllers/case/cancellation/reason-for-cancelling.controller.ts

```diff
@@ -7,8 +7,18 @@ import { validateReasonForCancelling } from './validation/validate-reason-for-ca
 import api from '../../../api';
 import { canSubmissionTypeBeCancelled } from '../../helpers';
 
-export type GetReasonForCancellingRequest = CustomExpressRequest<{ params: { _id: string } }>;
-export type PostReasonForCancellingRequest = CustomExpressRequest<{ params: { _id: string }; reqBody: { reason: string } }>;
+export type GetReasonForCancellingRequest = CustomExpressRequest<{ params: { _id: string }; query: { status?: string } }>;
+export type PostReasonForCancellingRequest = CustomExpressRequest<{ params: { _id: string }; query: { status?: string }; reqBody: { reason: string } }>;
+
+/**
+ * Gets the previous page URL to pass into the back link
+ *
+ * @param dealId - The deal ID
+ * @param status - 'change' if user comes from the check answers page, otherwise undefined
+ */
+const getPreviousPage = (dealId: string, status?: string): string => {
```

Comment:
Can we rename this to `getPreviousPageUrl` and move into a helpers directory please, so we have the same approach as @MarRobSoftwire 's recent PR

Looks like we might need to move the existing `getPreviousPageUrl` helper into a child directory so it's clear, maybe something like

`/controllers/case/cancellation/helpers/X-page/get-previous-page-url.ts`
`/controllers/case/cancellation/helpers/Y-page/get-previous-page-url.ts`

What do you think @BethThomas141 @MarRobSoftwire ?

Reply:
I've renamed it to `getPreviousPageUrl` thanks! I'm not sure if it's worth moving each of these out into a child directory - each of these functions is different in the 3 respective files, and only used in one place. 
The logic in these functions is also a lot simpler than in the case that Mark made a separate helper function for. 

I'd prefer to reduce file numbers and keep these at the top of these files (instead of making an additional 3 directories and 3 files).  

Would that be ok? Let me know what you think!

Reply:
Thank you both, agree on the points

Thing is, if they're moved to controller specific directories with a helpers directory - there's a lot of repeated code. All the new `getPreviousPageUrl` instances (in this PR at least), do exactly the same thing, except they have a different sub string at the end.

We could pass e.g a `route` param (for lack of a beter name) to a generic, DRY `getPreviousPageUrl`.

How about having this in a parent `/controllers/case/cancellation/helpers/get-previous-page-url` structure?

Then, the controllers become simpler and the controller tests just need to call `getPreviousPageUrl`, since we'd have specific unit tests for `getPreviousPageUrl`

Would that work / any other ideas? What do you think?

Reply:
@ttbarnes Sure! Have pulled it out into the same `/controllers/case/cancellation/helpers/get-previous-page-url` helpers file that Mark's one lives in. Let me know what you think.

Reply:
Nice one @BethThomas141 👏